titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/PersistentVolumeClaimGcController.java

```diff
@@ -107,20 +106,11 @@ private boolean gcPersistentVolumeClaim(V1PersistentVolumeClaim pvc) {
             // If the PVC is deleted while still in use by a pod (though that is not expected), the PVC
             // will not be removed until no pod is using it.
             // https://kubernetes.io/docs/concepts/storage/persistent-volumes/#storage-object-in-use-protection
-            kubeApiFacade.getCoreV1Api().deleteNamespacedPersistentVolumeClaim(
-                    volumeClaimName,
-                    DEFAULT_NAMESPACE,
-                    null,
-                    null,
-                    0,
-                    null,
-                    null,
-                    null
-            );
+            kubeApiFacade.deleteNamespacedPersistentVolumeClaim(DEFAULT_NAMESPACE, volumeClaimName);
             logger.info("Successfully deleted persistent volume claim {}", formatPvcEssentials(pvc));
             return true;
-        } catch (ApiException e) {
-            if (!e.getMessage().equalsIgnoreCase(NOT_FOUND)) {
+        } catch (KubeApiException e) {
+            if (e.getErrorCode() != KubeApiException.ErrorCode.NOT_FOUND) {
```

Comment:
if I read this code correctly, shouldn't this be an `==` comparison? If I'm right, this bug was already present before your change.

Reply:
Good catch. I kept the original logic which was also incorrect.